src/main/java/com/rultor/profiles/Profiles.java

```diff
@@ -73,18 +79,77 @@ public Profile fetch(final Talk talk) throws IOException {
             if (xml.nodes("/talk/wire").isEmpty()) {
                 profile = Profile.EMPTY;
             } else {
-                profile = Profiles.fetch(xml);
+                profile = this.fetch(xml);
             }
         }
         return profile;
     }
 
     /**
+     * Merge profile from master and fork. Merged profile must be profile
+     * from fork, but all lists for commanders and architects must be taken
+     * from master branch.
+     * @param master Profile from master branch
+     * @param fork Profile from fork branch
+     * @return Merged profile
+     * @todo #1039:30min Modify method implementation: merged profile is a
+     *  `.rultor.yml` form result of real `git merge` of master branch and
+     *  fork branch. We need real `git merge` result here. See some details
+     *  here: https://github.com/yegor256/rultor/pull/1064/files#r56796959
+     */
+    public Profile merged(final Profile master, final Profile fork) {
+        return master;
+    }
+
+    /**
+     * Validate merged profile: merged profile must have no changes in architect
+     * and commander sections (full match with master profile).
+     * @param master Profile from master branch
+     * @param merged Result of merging master and fork profiles
+     * @return Validated profile
+     * @throws IOException If fails
+     */
+    public Profile validated(final Profile master, final Profile merged)
+        throws IOException {
+        final String[] security = {"architect", "commander"};
+        for (final String section : security) {
+            if (!Arrays.equals(
+                section(master, section),
+                section(merged, section)
+            )) {
+                throw new Profile.ConfigException(
```

Comment:
@original-brownbear I'm not sure. I think this exception will be maintained here: 

https://github.com/xupyprmv/rultor/blob/%231039/src/main/java/com/rultor/Routine.java#L129